core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -1261,4 +1265,13 @@ public void setNonChMaxWaypointDistance(int nonChMaxWaypointDistance) {
         this.nonChMaxWaypointDistance = nonChMaxWaypointDistance;
     }
 
+    public void setSpatialRuleLookup(SpatialRuleLookup spatialRuleLookup) {
```

Comment:
Hmmh, I do not like this procedure much. We should not provide a setter here, if this is not an additional thing for GraphHopper (infact GraphHopper.spatialRuleLookup is unused).

Instead the user should/could do `graphHopper.getEncodingManager().getEncoder("generic").setXY` ... or maybe independent on generic and just `graphHopper.getEncodingManager().setXY`?